earthfile2llb/converter.go

```diff
@@ -1691,6 +1692,19 @@ func (c *Converter) internalRun(ctx context.Context, opts ConvertRunOpts) (pllb.
 	}
 	if !opts.Locally {
 		// Debugger.
+		err := c.opt.LLBCaps.Supports(solverpb.CapExecMountSock)
+		if err != nil {
+			_, isCapError := err.(*apicaps.CapError)
+			if !isCapError {
+				c.opt.Console.Warnf("failed to check LLBCaps for CapExecMountSock: %v", err) // keep going
+			} else if opts.Interactive {
+				// only display warning when user supplies --interactive to an older buildkit version
+				c.opt.Console.Warnf("interactive debugger requires a newer version of buildkit: %v", err)
```

Comment:
So this breaks compatibility... but then again nobody is using the debugger in remote buildkit / satellite context today. And for local buildkit, the versions are in lock-step, so it's not an issue.

Reply:
my thinking exactly!